### PR TITLE
[15.0] Migration of account_operating_unit

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

Migration PR for module account_operating_unit.

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### Comments

This module depends on 
https://github.com/OCA/operating-unit/pull/418
https://github.com/OCA/operating-unit/pull/419

### CODE REVIEW

**Review by olveracuenca — APPROVED** [review]

**Review by AaronHForgeFlow — COMMENTED** [review]
I can't create in ronboat an account.journal for a different operating unit than "Main Operating Unit", it seems that is not correct, because I am allowed to access all operating units

**Comment by alan196** [comment]
@AaronHForgeFlow I had the same feeling at the beginning, but it's a strange behaviour of many2one fields.

You can see a screen recording from Runboat instance.
account_operating_unit

**Comment by AaronHForgeFlow** [comment]
@alan196 you are right :)

However, when I pay a bill for B2BOU using B2COU and I set the feature in hte company "OU are self-balanced" I cannot see the journal entry that balances the OU created:

**Review by BT-apadron — APPROVED** [review]

**Review by BT-pcavero — APPROVED** [review]

**Comment by BT-pcavero** [comment]
Greetings @alan196!
What is the current status of this migration?

**Comment by alan196** [comment]
I have it working in a production database, it works, but as @AaronHForgeFlow comments, it doesn't work for OU are self-balanced accounts, at this moment I don't have enough time to look at this PR, you can take this PR and fix that issues if you want I can close this PR

**Comment by AaronHForgeFlow** [comment]
I wish I had the time also to fix this, I will try to help if I manage to have some time in a short term, but I cannot ensure that.

**Comment by AaronHForgeFlow** [comment]
Need to check if this is needed https://github.com/OCA/operating-unit/pull/457

**Comment by olveracuenca** [comment]
I am very well aware :+1:

File: account_operating_unit/models/account_journal.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountJournal(models.Model):
+    _inherit = "account.journal"
+
+    operating_unit_id = fields.Many2one(
+        comodel_name="operating.unit",
+        domain="[('user_ids', '=', uid)]",
```
Comment (Saran440) [inline account_operating_unit/models/account_journal.py:14]:
```suggestion
```

File: account_operating_unit/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            self._context.get("default_move_type", False)
+            and self._context.get("default_move_type") != "entry"
+        ):
+            return self.env["res.users"].operating_unit_default_get()
+        return False
+
+    operating_unit_id = fields.Many2one(
+        comodel_name="operating.unit",
+        default=_default_operating_unit_id,
+        domain="[('user_ids', '=', uid)]",
```
Comment (Saran440) [inline account_operating_unit/models/account_move.py:84]:
```suggestion
```

**Review by Saran440 — COMMENTED** [review]
I have question, why we add domain in `operating_unit_id` ?
If it not use we should delete it all. I test with delete domain. it's work.

User has 2 OU


User has 3 OU

**Comment by Saran440** [comment]
@alan196 @AaronHForgeFlow
Hello, i think this bug because delete code compute field https://github.com/OCA/operating-unit/pull/420/commits/da6875f359e7b749d0e98d24bb577634f6271e2d#diff-7b31f2a05a98aa289b2276c57450a4d8d549ae1ea7a4ece9f1e61fe36eb67e3bL12

So, I update code and test script here > https://github.com/OCA/operating-unit/pull/522
Do you want to cherry-pick to this PR?

**Comment by AaronHForgeFlow** [comment]
Superseded by https://github.com/OCA/operating-unit/pull/522

**Pull request closed without merging**